utopia-remix/app/env.server.ts

```diff
@@ -39,19 +41,40 @@ export type BrowserEnvironment = {
 }
 
 export const BrowserEnvironment: BrowserEnvironment = {
-  EDITOR_URL: mustEnv('REACT_APP_EDITOR_URL'),
-  UTOPIA_CDN_URL: optionalEnv('UTOPIA_CDN_URL', ''),
+  EDITOR_URL: mustEnvOrLocalFallback('REACT_APP_EDITOR_URL', 'http://localhost:8000'),
+  UTOPIA_CDN_URL: mustEnvOrLocalFallback('UTOPIA_CDN_URL', 'http://localhost:8000'),
 }
 
-function mustEnv(key: string): string {
-  const value = process.env[key]
-  if (value == null) {
-    throw new Error(`missing required environment variable ${key}`)
+/**
+ * If the APP_ENV variable is set, return that.
+ * If it's not set, try to guess it from the NODE_ENV.
+ * If everything else fails, throw an error, because this should _never_ happen.
+ */
+function getAppEnv(): AppEnv {
+  const env = process.env['APP_ENV'] ?? ''
+  if (env !== '') {
+    return env
+  }
+
+  switch (process.env.NODE_ENV) {
+    case 'development':
+      return 'local'
+    case 'production':
+      return 'prod'
+    case 'test':
+      return 'test'
+    default:
+      throw new Error('cannot determine app environment')
   }
-  return value
 }
 
-function optionalEnv(key: string, defaultValue: string): string {
-  const value = process.env[key]
-  return value ?? defaultValue
+function mustEnvOrLocalFallback(key: string, localFallback: string) {
+  const value = process.env[key] ?? ''
+  if (value !== '') {
+    return value
+  }
```

Comment:
I think the `?? ''` here is not needed

```suggestion
  const value = process.env[key]
  if (value != null) {
    return value
  }
```

Reply:
or maybe even 
```suggestion
  if (key in process.env) {
    return process.env[key]
  }
```

Reply:
the lookup can give back an undefined tho, _or_ you can pass an env var that is defined but is an empty string